### PR TITLE
Adding missing arguments to fastp

### PR DESCRIPTION
**Pull request by RZ9082**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

Add `--merge`, `--include_unmerged`, and test-files  to fastp

Fastp can now be used for merging forward and reverse reads. When choosing 'Paired' or 'Paired Collection' the user will have the option to select `--merge` which will produce the following outputs:
* **Merged reads**
* **Unmerged filtered reads1** and **Unmerged filtered reads2** are the unable to merge reads, but they pass all the filters.
* **Unmerged unfiltered reads1** are unmerged reads, where only read1 passes the filters. 
* **Unmerged unfiltered reads2** are unmerged reads, where only read2 passes the filters.

Furthermore if the user selects the `--merge` option, the `--include_unmerged` option will be also chooseable, which will include all the outputs mentioned above in the **Merged reads** file.

### CODE REVIEW

**Comment by bgruening** [comment]
Can you please take this one over into your PR as well https://github.com/galaxyproject/tools-iuc/pull/6443/files

File: tools/fastp/fastp.xml [hunk 1]
```diff
@@ -41,6 +41,7 @@
     #end if
 #end if
 
+cp $c1 galaxy.json &&
```
Comment (bernt-matthias) [inline tools/fastp/fastp.xml:44]: Config file needs quotes.

**Review by bernt-matthias — COMMENTED** [review]
Can you explain a bit, what has changed here?

**Comment by RZ9082** [comment]
> Can you explain a bit, what has changed here?
I added an explanation to the first comment.

File: tools/fastp/fastp.xml [hunk 2]
```diff
@@ -54,14 +55,25 @@ fastp
...
 
-#if str($single_paired.single_paired_selector).startswith('paired'):
+## Merge reads
+
+#if str($single_paired.single_paired_selector).startswith('paired')
     -I '$in2_name'
-    -O second${ext}
+    #if $single_paired.merge_reads.merge
+        #if $single_paired.merge_reads.include_unmerged
+            $single_paired.merge_reads.merge --merged_out '$merged_reads' $single_paired.merge_reads.include_unmerged
```
Comment (bernt-matthias) [inline tools/fastp/fastp.xml:65]: You could move `$single_paired.merge_reads.merge` out of the if-else (since it appears in both).

Comment (RZ9082) [reply tools/fastp/fastp.xml:65]: > I think you want to have --merged_out '$merged_reads' --unpaired1 '$unpaired1' --unpaired2 '$unpaired2'`in any case and`--out1 '$unmerged_out1' --out2 '$unmerged_out2'`only if`include_unmerged`.

I followed the logic explained [here](https://github.com/OpenGene/fastp/blob/master/README.md#merge-paired-end-reads)
When using `--merge` you get out1, out2, unpaired1, and unpaired2. If you use it in combination with `--include_unmerged`, all the outputs before will be directed into the merged reads file.

File: tools/fastp/fastp.xml [hunk 3]
```diff
@@ -54,14 +55,25 @@ fastp
...
+        #else
+            $single_paired.merge_reads.merge --merged_out '$merged_reads' --out1 '$unmerged_out1' --out2 '$unmerged_out2' --unpaired1 '$unpaired1' --unpaired2 '$unpaired2'
```
[the remark this answers is not in the document]
Comment (RZ9082) [reply tools/fastp/fastp.xml:67]: sure, will do.

File: tools/fastp/fastp.xml [hunk 4]
```diff
@@ -315,31 +348,46 @@ mv first${ext} '${out1}'
 
         <section name="output_options" title="Output Options" expanded="False">
             <param name="report_html" type="boolean" truevalue="True" falsevalue="False" checked="True" label="Output HTML report" help="fastp provides a QC report for the data Before and After filtering within a single HTML page, which enables comparison of the quality statistics changed by the preprocessing step directly"/>
-            <param name="report_json" type="boolean" truevalue="True" falsevalue="False" checked="False" label="Output JSON report" help="The JSON report contains all the data visualized in the HTML report. The format of the JSON report is manually optimized to be easily readable by humans and is compatible with MultiQC"/>
+            <param name="report_json" type="boolean" truevalue="True" falsevalue="False" checked="True" label="Output JSON report" help="The JSON report contains all the data visualized in the HTML report. The format of the JSON report is manually optimized to be easily readable by humans and is compatible with MultiQC"/>
```
[the remark this answers is not in the document]
Comment (RZ9082) [reply tools/fastp/fastp.xml:351]: This was suggested in another [PR](https://github.com/galaxyproject/tools-iuc/pull/6443). Basically to be able in the future to replace fastqc with fastp.

File: tools/fastp/fastp.xml [hunk 5]
```diff
@@ -373,7 +449,8 @@ mv first${ext} '${out1}'
             <param name="in1" ftype="fastq" value="R1.fq"/>
```
Comment (bernt-matthias) [inline tools/fastp/fastp.xml:449]: Still need to forbid fastq (and explicitly list the allowed fastq types, I guess fastqsanger and fastqillumina + .gz)

**Review by bernt-matthias — APPROVED** [review]

**Comment by RZ9082** [comment]
Thanks :)

**Pull request merged**